app/graphql/mutations/time_records/create.rb

```diff
@@ -0,0 +1,26 @@
+module Mutations
+  module TimeRecords
+    class Create < Mutations::BaseMutation
+      graphql_name 'TimeRecordCreate'
+
+      argument :data, Types::Inputs::TimeRecordData, required: true
+
+      field :time_record, Types::Models::TimeRecord, null: false
+
+      def resolve(data:)
+        authorize('time_record')
+        time_record = create_record(data)
+        return { time_record: time_record }
+      rescue ActiveRecord::RecordInvalid => e
+        GraphQL::ExecutionError.new(e)
+      end
+
+      private
+
+      def create_record(data)
+        record_params = data.to_h.merge(time_start: Time.now)
```

Comment:
In tests I found an example of request. In this request spentTime is defined. 
So record will be created for Time.now and in will end in the future. 
But what if I send another record with spentTime in several minutes, when several minutes < that spentTime of first record? We will have to tasks that intersects. 

Also the force rewriting of time_start here it's a bad thing. Maybe Time.now should be default value. Or ability to send spend_time via creating should be forbidden.

Maybe it's all species of client, but than it should declare all information about record, not backend. And backend should validate intersects of records, I think.

Reply:
I think we clarified this issue. We also can complex the logic of tracking time later.